src/rpc/claimtrie.cpp

```diff
@@ -22,143 +24,177 @@ uint160 ParseClaimtrieId(const UniValue& v, const std::string& strName)
+        if (it->second.claims.empty()) continue;
+
+        UniValue claims(UniValue::VARR);
+        for (std::vector<CClaimValue>::iterator itClaims = it->second.claims.begin(); itClaims != it->second.claims.end(); ++itClaims) {
+            UniValue claim(UniValue::VOBJ);
+            claim.push_back(Pair("claimId", itClaims->claimId.GetHex()));
+            claim.push_back(Pair("txid", itClaims->outPoint.hash.GetHex()));
+            claim.push_back(Pair("n", (int)itClaims->outPoint.n));
+            claim.push_back(Pair("amount", ValueFromAmount(itClaims->nAmount)));
+            claim.push_back(Pair("height", itClaims->nHeight));
+            const CCoins* coin = coinsCache.AccessCoins(itClaims->outPoint.hash);
+            if (!coin) {
+                LogPrintf("%s: %s does not exist in the coins view, despite being associated with a name\n",
+                    __func__, itClaims->outPoint.hash.GetHex());
+                claim.push_back(Pair("error", "No value found for claim"));
+            } else if (!coin->IsAvailable(itClaims->outPoint.n)) {
+                LogPrintf("%s: the specified txout of %s appears to have been spent\n", __func__, itClaims->outPoint.hash.GetHex());
+                claim.push_back(Pair("error", "Txout spent"));
+            } else {
+                int op;
+                std::vector<std::vector<unsigned char> > vvchParams;
+                if (!DecodeClaimScript(coin->vout[itClaims->outPoint.n].scriptPubKey, op, vvchParams)) {
+                    LogPrintf("%s: the specified txout of %s does not have an claim command\n", __func__, itClaims->outPoint.hash.GetHex());
                 }
-                claims.push_back(claim);
+                std::string sValue(vvchParams[1].begin(), vvchParams[1].end());
+                claim.push_back(Pair("value", sValue));
             }
-            node.push_back(Pair("claims", claims));
-            ret.push_back(node);
+            claims.push_back(claim);
         }
+
+        UniValue node(UniValue::VOBJ);
+        node.push_back(Pair("name", it->first));
+        node.push_back(Pair("claims", claims));
+        ret.push_back(node);
     }
     return ret;
 }
 
 UniValue getclaimtrie(const UniValue& params, bool fHelp)
 {
-    if (fHelp || params.size() > 0)
+    if (fHelp || params.size() > 1)
         throw std::runtime_error(
             "getclaimtrie\n"
             "Return the entire name trie.\n"
             "Arguments:\n"
-            "None\n"
+            "1. \"blockhash\"   (string, optional) the hash of the block\n"
+            "                                         which is the active\n"
+            "                                         in the chain. If\n"
+            "                                         none is given, \n"
+            "                                         the latest block\n"
+            "                                         will be used.\n"
             "Result: \n"
```

Comment:
how about:
"get the claim trie at the block specified by this block hash. If none is given, the latest active block will be used"